### PR TITLE
IA-1644 lambda-edge to handle redirects

### PR DESCRIPTION
**Pull request by jsmigovsky**
Need to wire in the behaviors @brooksa321 and @USSBA/ias 
Waiting until NGINX paths are moved to CloudFront behaviors.

### CODE REVIEW

File: cloudformation/templates/sbagov/cloudfront.yml [hunk 1]
```diff
@@ -949,6 +952,90 @@ Resources:
...
+              default:
+                // A match was found, redirecting viewer accordingly
+                var location = result.Items[0].NewUrl.S;
+                if (result.Count > 1) {
+                  console.log(`Duplicate Records for URI ${request.uri}`);
+                }
+                console.log(`Relocating from ${request.uri} to ${location}`);
+                var redirect = {
+                  status: '302',
+                  statusDescription: 'Moved Permanently',
```
[the remark this answers is not in the document]
Comment (jsmigovsky) [reply cloudformation/templates/sbagov/cloudfront.yml:1037]: changed to 301

File: cloudformation/templates/sbagov/cloudfront.yml [hunk 2]
```diff
@@ -949,6 +952,119 @@ Resources:
...
+            console.log(`Path = '${loc.path}'`);
+            console.log(`QueryString = '${loc.queryString}'`);
+
+            // see if we have a match in the dynamodb table
+            var result = await query.find(loc.path);
+
+            switch(result.Count) {
+              case 0:
+                // continue to origin
+                // note: we may need to strip off the forward slash and redirect
```
Comment (bushong1) [inline cloudformation/templates/sbagov/cloudfront.yml:1016]: @brooksa321 any thoughts on this?

File: cloudformation/templates/sbagov/cloudfront.yml [hunk 3]
```diff
@@ -949,6 +952,119 @@ Resources:
...
+                console.log(`redirect not found for ${loc.path}`);
+                callback(null, response);
+                break;
+              default:
+                // mathing record was found
```
Comment (bushong1) [inline cloudformation/templates/sbagov/cloudfront.yml:1021]: `matching`

File: cloudformation/templates/sbagov/cloudfront.yml [hunk 4]
```diff
@@ -949,6 +952,119 @@ Resources:
...
+                var location = result.Items[0].NewUrl.S;
+
+                // if the redirect location is the same as the request uri continue to origin
+                if (location === loc.path) {
+                  console.log(`A redirect to ${location} would have caused an infinate loop`);
```
Comment (bushong1) [inline cloudformation/templates/sbagov/cloudfront.yml:1026]: `infinite`

File: cloudformation/templates/sbagov/cloudfront.yml [hunk 5]
```diff
@@ -949,6 +952,119 @@ Resources:
...
+                  callback(null, response);
+                  break;
+                }
+                console.log(`Redirecting from ${[loc.path, loc.queryString].join('')} to ${[location, loc.queryString].join('')}`);
+                var redirect = {
+                  status: '301',
```
Comment (bushong1) [inline cloudformation/templates/sbagov/cloudfront.yml:1032]: I think we want these redirects to be temporary.  Any thoughts @brooksa321 ?

**Pull request merged**